TESTS/Arduino/AnalogIn/AnalogIn.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if !DEVICE_ANALOGIN
+  #error [NOT_SUPPORTED] AnalogIn not supported on this platform, add 'DEVICE_ANALOGIN' definition to your platform.
+#endif
+
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "ci_test_config.h"
+
+using namespace utest::v1;
+
+// Template to set one Analog pin as input and then cycle through the rest as outputs.
+// As you turn more pins on the voltage on the ain pin will go up.
+template <PinName ain_pin> 
+void AnalogInput_Test(){
+  AnalogIn ain(ain_pin);
+  DEBUG_PRINTF("Tested pin %d for Arduino analog in capabilities\n", ain_pin);
```

Comment:
Global comment: from my side, I would prefer a debug printf as %#x instead of %d for pin number